source/CCustomOpcodeSystem.cpp

```diff
@@ -835,7 +838,9 @@ namespace CLEO {
 		bool savedNotFlag;
 		static const size_t store_size = 0x400;
 		static ScmFunction *Store[store_size];
-		static size_t allocationPlace;			// contains an index of last allocated object
+		static size_t allocationPlace; // contains an index of last allocated object
+		std::string savedScriptFileDir; // modules switching
+		std::string savedScriptFileName; // modules switching

```

Comment:
probably makes sense to combine all data that we save from the current script into a new struct called Context.

Reply:
Context constructor would take CCustomScript and grab necessary fields. It should keep the ScmFunction implementation cleaner

Reply:
Started implementing it but turned out to be a bit redundant. Seems like instead of creating new class and copying declarations of all properties instance CCustomScript itself should be used. Problem is it's properties are all private and getter setters should be used. Anyway this can be solved by friend class.
This Context or ScriptState/Snapshot class should have store() and restore() functions. Problem is that scmFunction does not restoring all script properties, only few of them. So we will end with illusion of general purpose class that in fact is just repacked contents of current scmFunction functionality.

Reply:
Maybe current ScmFunction should just be refactored to be better named and logically grouped.